backend/dal.keycloak/Partials/PimsKeycloakUserService.cs

```diff
@@ -180,7 +180,7 @@ public async Task<Entity.AccessRequest> UpdateAccessRequestAsync(Entity.AccessRe
 
             this._user.ThrowIfNotAuthorized(Permissions.AdminUsers, Permissions.AgencyAdmin);
             var accessRequest = _pimsAdminService.User.GetAccessRequest(entity.Id);
-            if (!accessRequest.IsGranted != true && entity.IsGranted == true)
+            if (accessRequest.Status != Entity.AccessRequestStatus.APPROVED && entity.Status == Entity.AccessRequestStatus.APPROVED)
```

Comment:
This code will execute only if status from the database is not APPROVED and the next state is set to APPROVE (admin intends to approve the access request)

I can change this code so that it checks if the current status is ON_HOLD and the next state is APPROVED

Reply:
So the current logic allows them to go from Denied to Approved?  If so, does everything work correctly on the management page?  I think if it works, then we keep it as you have it.